.github/workflows/e2e-instances.yml

```diff
@@ -0,0 +1,302 @@
+name: Testing
+
+on: [pull_request]
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  prepare_variables:
+    runs-on: ubuntu-22.04
+    outputs:
+      POOL_NAME: ${{ steps.generate.outputs.POOL_NAME }}
+      POOL_INSTANCE: ${{ steps.generate.outputs.POOL_INSTANCE }}
+      BASE_URL: ${{ steps.generate.outputs.BASE_URL }}
+      API_URI: ${{ steps.generate.outputs.API_URI }}
+      BACKUP_ID: ${{ steps.backup.outputs.BACKUP_ID }}
+      BACKUP_VER: ${{ steps.backup.outputs.BACKUP_VER }}
+      BACKUP_NAME: ${{ steps.backup.outputs.BACKUP_NAME }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Inject slug/short variables
+        uses: rlespinasse/github-slug-action@v4
+
+      - name: Generate
+        id: generate
+        run: |
+          echo "BASE_URL=${{ env.GITHUB_HEAD_REF_SLUG_URL }}.dashboard.saleor.rocks" >> $GITHUB_OUTPUT
+          echo "API_URI=https://${{ env.GITHUB_HEAD_REF_SLUG_URL }}.staging.saleor.cloud/graphql/" >> $GITHUB_OUTPUT
+          echo "POOL_NAME=${{ env.GITHUB_HEAD_REF_SLUG_URL }}" >> $GITHUB_OUTPUT
+          echo "POOL_INSTANCE=https://${{ env.GITHUB_HEAD_REF_SLUG_URL }}.staging.saleor.cloud" >> $GITHUB_OUTPUT
+
+      - name: Saleor login
+        uses: ./.github/actions/cli-login
+        with:
+          token: ${{ secrets.STAGING_TOKEN }}
+
+      - name: Obtain backup id
+        id: backup
+        env:
+          SALEOR_CLI_ENV: staging
+          BACKUP_NAME: snapshot-automation-tests
+        run: |
+          BACKUPS=$(npx saleor backup list --name=snapshot-automation-tests --latest --json)
+          BACKUP_ID=$(echo $BACKUPS | jq -r '.[0].key')
+          BACKUP_VER=$(echo $BACKUPS | jq -r '.[0].saleor_version')
+          BACKUP_NAME=$(echo $BACKUPS | jq -r '.[0].name')
+
+          echo "BACKUP_ID=$BACKUP_ID" >> $GITHUB_OUTPUT
+          echo "BACKUP_VER=$BACKUP_VER" >> $GITHUB_OUTPUT
+          echo "BACKUP_NAME=$BACKUP_NAME" >> $GITHUB_OUTPUT
+
+      - name: Print annotations
+        run: |
+          echo "::notice title=BASE_URL::${{ steps.generate.outputs.BASE_URL }}"
+          echo "::notice title=API_URI::${{ steps.generate.outputs.API_URI }}"
+          echo "::notice title=POOL_NAME::${{ steps.generate.outputs.POOL_NAME }}"
+          echo "::notice title=POOL_INSTANCE::${{ steps.generate.outputs.POOL_INSTANCE }}"
+          echo "::notice title=SNAPSHOT::backup_id=${{ steps.backup.outputs.BACKUP_ID }}, version=${{ steps.backup.outputs.BACKUP_VER }}, name=${{ steps.backup.outputs.BACKUP_NAME }}"
```

Comment:
Follow best practices https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#understanding-the-risk-of-script-injections